### DISCUSSION TITLE
/login/oauth/access_token should return JSON, not Form URL Encoded

### DISCUSSION THREAD

**Discussion by davedelong** [community/community#116814] [Apps, API and Webhooks]
### Select Topic Area

Bug

### Body

[RFC 6749 § 5.1](https://datatracker.ietf.org/doc/html/rfc6749#section-5.1) says:

> The authorization server issues an access token and optional refresh token, and constructs the response by adding the following parameters to the entity-body of the HTTP response with a 200 (OK) status code: …
> 
> The parameters are included in the entity-body of the HTTP response using the "application/json" media type as defined by [[RFC4627](https://datatracker.ietf.org/doc/html/rfc4627)].  The parameters are serialized into a JavaScript Object Notation (JSON) structure by adding each parameter at the highest structure level.

However, GitHub's implementation is non-compliant, because [it returns values in `application/x-www-form-urlencoded` format](https://docs.github.com/en/apps/oauth-apps/building-oauth-apps/authorizing-oauth-apps#2-users-are-redirected-back-to-your-site-by-github). This makes it impossible to use standards-compliant OAuth libraries to access the GitHub API (such as the "AppAuth-iOS" library), because even the option of adding a custom `Accept:` header is out of spec for OAuth 2. Requests to add support to libraries are [repeatedly](https://github.com/openid/AppAuth-iOS/pull/206) [denied](https://github.com/openid/AppAuth-iOS/pull/375). The _official_ recommendation from these library providers is to fork their repository and make modifications to it.

GitHub should change their OAuth implementation to return `application/json` content.

**Comment by giovannivicentin** [comment]


**Reply by ghostinhershell** [reply]
Hi @davedelong, 

We’ve clarified our stance on using generative AI tools like ChatGPT within our Community via [this announcement](https://github.com/orgs/community/discussions/69006). Please review the guidelines to ensure your post meets them as failure to adhere to those rules can result in action taken by our moderator team.  You can read our updated [Code of Conduct](https://github.com/community/community/blob/baa5cd9020ef3f2bbf86dca1191d9c90f9b0822d/CODE_OF_CONDUCT.md#reasonable-use-of-ai-generated-content) and the [announcement](https://github.com/orgs/community/discussions/69006) for more details. Thank you for helping us maintain an authentic and beneficial space for everyone.

**Reply by davedelong** [reply]
@ghostinhershell  This reply is not mine; that's why I reported it as spam.
